core/vm/contracts_suave.go

```diff
@@ -220,12 +220,20 @@ func (s *suaveRuntime) doHTTPRequest(request types.HttpRequest) ([]byte, error)
 	}
 
 	var allowed bool
-	for _, allowedDomain := range s.suaveContext.Backend.ExternalWhitelist {
-		if allowedDomain == "*" || allowedDomain == parsedURL.Hostname() {
-			allowed = true
-			break
+	// resolve dns if possible
+	if endpoint, ok := s.suaveContext.Backend.DnsRegistry[request.Url]; ok {
+		request.Url = endpoint
+		allowed = true
+	} else {
+		// check if the domain is allowed
```

Comment:
```suggestion
parsedURL, err := url.Parse(request.Url)
if err != nil {
     panic(err)
}
```
You do not have to parse the url if its a dns domain.